lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -745,9 +745,15 @@ fn create_segment_id_tracker(
         };
 
         if use_rocksdb_mutable_tracker {
-            return Ok(sp(IdTrackerEnum::RocksDbIdTracker(
-                create_rocksdb_id_tracker(db_builder.require()?)?,
-            )));
+            let id_tracker = create_rocksdb_id_tracker(db_builder.require()?)?;
+
+            // Actively migrate RocksDB based ID tracker into mutable ID tracker
+            if feature_flags().migrate_rocksdb_id_tracker {
+                let id_tracker = migrate_rocksdb_id_tracker_to_mutable(id_tracker, segment_path)?;
+                return Ok(sp(IdTrackerEnum::MutableIdTracker(id_tracker)));
+            }
+
+            return Ok(sp(IdTrackerEnum::RocksDbIdTracker(id_tracker)));
```

Comment:
Note: this is the only place left where we load the RocksDB based ID tracker. It is loaded if it still exists on disk.

Since this is the only site, we can migrate the ID tracker here.